apps/hash-graph/lib/graph/src/shared/identifier/time/bound.rs

```diff
@@ -0,0 +1,263 @@
+use std::ops::Bound;
+
+use derivative::Derivative;
+use serde::{Deserialize, Serialize};
+use utoipa::{
+    openapi,
+    openapi::{RefOr, Schema},
+    ToSchema,
+};
+
+use crate::{
+    identifier::time::{axis::TemporalTagged, Timestamp},
+    interval::IntervalBound,
+};
+
+#[derive(Derivative, Serialize, Deserialize, ToSchema)]
+#[derivative(
```

Comment:
Why do we need to use `derivative` for these again?

Reply:
The `#[derive(Debug)]` (and others as well) will generate `impl<A: Debug> ...` instead of `impl<A> ...`.